### PR TITLE
Fixing Contact urls for linked authors

### PR DESCRIPTION
**Pull request by Hackwar**
Pull Request for Issue #12603.
### Summary of Changes

Used the right method to generate that URL.
### Testing Instructions

Create a contact for the super user
Create some articles
Create a menu item to display one or multiple articles, do NOT make it the home page.
Create a menu item displaying the Contact category to which belongs the super user contact: alias is english-contacts in this case.
Do NOT create a menu item displaying the single contact.
Display the article in frontend.
Click on the Super User link in the article intro
See that the link is not correct.
Apply patch and see that the link is correct.
### Documentation Changes Required

none

### CODE REVIEW

File: plugins/content/contact/contact.php [hunk 1]
```diff
@@ -57,15 +57,13 @@ public function onContentPrepare($context, &$row, $params, $page = 0)
...
 
-		if ($row->contactid)
+		if ($contact)
 		{
-			$needle = 'index.php?option=com_contact&view=contact&id=' . $row->contactid;
-			$menu = JFactory::getApplication()->getMenu();
-			$item = $menu->getItems('link', $needle, true);
-			$link = $item ? $needle . '&Itemid=' . $item->id : $needle;
-			$row->contact_link = JRoute::_($link);
+			require_once JPATH_ROOT . '/components/com_contact/helpers/route.php';
```
Comment (mbabker) [inline plugins/content/contact/contact.php]: Can you load this with `JLoader::register()`?  Most of the include/require statements have been replaced with registering classes on the autoloader.

**Comment by Hackwar** [comment]
done

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on c5c20f1746bb5ee94a3a63a848d0c7e48ffab922<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12616">issues.joomla.org/tracker/joomla-cms/12616</a>.</sub>

**Comment by brianteeman** [comment]
I have updated the base branch to staging

@Hackwar please dont use the 3.7 branch

**Comment by infograf768** [comment]
This needs one more tester.

**Comment by Hackwar** [comment]
Please note that by changing the branch from 3.7.x to staging, an additional file with unrelated changes was added to this PR, which I did not touch. I'm only taking responsibility for the contact plugin.

**Comment by infograf768** [comment]
@Hackwar
Can you just redo the PR towards staging and delete this one?

**Comment by Hackwar** [comment]
Closing this one in favour of #12629

**Pull request closed without merging**